h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -267,6 +378,48 @@ public static Vec categoricalToInt(final Vec src) {
     return newVec;
   }
 
+  public static Vec intToDouble(final Vec src) {
+    Vec newVec = copyOver(src, Vec.T_NUM, null);
+
+    new MRTask() {
+      @Override
+      public void map(Chunk c) {
+        for (int i = 0; i < c._len; ++i) {
+          if (!c.isNA(i)) {
+            c.set(i, (c.at8(i)+EPS)*2.0-c.at8(i)-2.0*EPS);
```

Comment:
Why do we need to used the formula `(c.at8(i)+EPS)*2.0-c.at8(i)-2.0*EPS`?

Reply:
I have tried so many ways to trick the system to use a double chunk to store our data instead of an integer chunk.  This is the first one that I hit and worked.  If I do 1.0*c.at8(i), the final vector will still come out as integer.  I was trying to avoid digging into the chunk allocation stuff but may have to do it.

Reply:
**I know h2o-core very little** so this might be a very stupid question but `Vec.isInt` just checks the rollupStats (`isInt(){return rollupStats()._isInt; }`). And in rollup stats  the `_isInt` is `_isInt = ((long)d) == d;` and is reduced like this `_isInt &= rs._isInt;` so it looks to me that individual chunks don't know if they are double or long (we seem to store longs in double chunk (losslessly if the number is smaller that $2^{52}$
(https://github.com/h2oai/h2o-3/blob/99fa6ac74e2a8c24bd26fea5364f525c18a09544/h2o-core/src/main/java/water/fvec/Chunk.java#L411-L430)) and I think if the chunk is short/int/long then when we call `atd()` it gets converted to double (https://github.com/h2oai/h2o-3/blob/e976be5a406d28c4ad770bc1156ec3ffa940080d/h2o-core/src/main/java/water/fvec/CXIChunk.java#L110-L115). 

 So I think `_isInt` is just a flag that tells H2O that the value is long stored in a double  so converting to double could be as easy as adding to `Vec` class the following and calling it in this method:
 
```java
 public void asDouble(){
    assert _type == T_NUM;
    rollupStats()._isInt = false;
  }
```

The `vec.isInt()` is then used for determining type so it influences whether `at8()` or `atd()` is called and also the types reported by Frame (https://github.com/h2oai/h2o-3/blob/e5ef3cab8cbe171c8f5b45a613eb0c178a187a67/h2o-core/src/main/java/water/api/schemas3/FrameV3.java#L171-L175). I looked only at couple chunk subclasses but `atd()` looks safe to be used from the chunks specifically created to store integer values and of course for the double chunks as well.

So I think we don't really have to care how the data is stored (especially if it's stored losslessly) but we should just changed how it is used.

One possible issue with this approach is that when the `RollupStats` get recomputed the type will change again to `int`. Unfortunately, this can happen even with a Vec with one double value - if we remove it the type of the whole column changes to int.

I'm sure it's not as easy as I wrote but I don't know what am I missing so I'd be happy if you correct me here so I can learn more things about h2o core.

Reply:
Your method actually works.  Thanks!  Now I can focus on reviewing your PR!  Thank you Tomas